code/identity/src/test/java/com/adobe/marketing/mobile/identity/IdentityFunctionalTests.kt

```diff
@@ -2037,4 +2042,178 @@ class IdentityFunctionalTests {
+            countDownLatch.countDown()
+        }.`when`(mockedExtensionApi).dispatch(any())
+
+        val persistedData = capturePersistedData()
+
+        val pushToken = "D52DB39EEE21395B2B67B895FC478301CE6E936D82521E095902A5E0F57EE0B3"
+
+        identityExtension.processIdentityRequest(
+            Event.Builder(
+                "event",
+                "com.adobe.eventType.generic.identity",
+                "com.adobe.eventSource.requestContent"
+            ).setEventData(
+                mapOf(
+                    "pushidentifier" to pushToken
+                )
+            ).build()
+        )
+
+        countDownLatch.await()
+
+        assertEquals(pushToken, persistedData["ADOBEMOBILE_PUSH_IDENTIFIER"] as? String)
+        assertTrue(persistedData["ADOBEMOBILE_ANALYTICS_PUSH_SYNC"] as? Boolean ?: false)
+        assertTrue(persistedData["ADOBEMOBILE_PUSH_ENABLED"] as? Boolean ?: false)
+    }
+
+    @Test(timeout = 10000)
+    fun test_setPushIdentifier_null_dispatchesAnalyticsForIdentityRequest() {
+        val configuration = mapOf(
+            "experienceCloud.org" to "orgid",
+            "experienceCloud.server" to "test.com",
+            "global.privacy" to "optedin"
+        )
+        val identityExtension = initializeIdentityExtensionWithPreset(
+            FRESH_INSTALL_WITHOUT_CACHE,
+            configuration
+        )
+
+        val countDownLatch = CountDownLatch(1)
+        doAnswer { invocation ->
+            assertEquals(1, invocation.arguments.size)
+            val event: Event? = invocation.arguments[0] as Event?
+            assertEquals(EventType.ANALYTICS, event?.type)
+            assertEquals(EventSource.REQUEST_CONTENT, event?.source)
+            val eventData = event?.eventData
+            assertNotNull(eventData)
+            val jsonObject = JSONObject(eventData)
+            assertEquals("Push", jsonObject.getString("action"))
+            assertTrue(jsonObject.getBoolean("trackinternal"))
+            val jsonContextObject = jsonObject.getJSONObject("contextdata")
+            assertEquals("false", jsonContextObject.getString("a.push.optin"))
+            countDownLatch.countDown()
+        }.`when`(mockedExtensionApi).dispatch(any())
+
+        val persistedData = capturePersistedData()
+
+        identityExtension.processIdentityRequest(
+            Event.Builder(
+                "event",
+                "com.adobe.eventType.generic.identity",
+                "com.adobe.eventSource.requestContent"
+            ).setEventData(
+                mapOf(
+                    "pushidentifier" to null
+                )
+            ).build()
+        )
+
+        countDownLatch.await()
+
+        assertNull(persistedData["ADOBEMOBILE_PUSH_IDENTIFIER"] as? String)
+        assertTrue(persistedData["ADOBEMOBILE_ANALYTICS_PUSH_SYNC"] as? Boolean ?: false)
+        assertFalse(persistedData["ADOBEMOBILE_PUSH_ENABLED"] as? Boolean ?: true)
+    }
+    @Test(timeout = 10000)
```

Comment:
Can you remove the timeout param? The test does not wait on any asynchronous operation.